app/models/task.py

```diff
@@ -1,5 +1,47 @@
 from app import db
 
-
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    task_id = db.Column(db.Integer, primary_key=True, autoincrement = True)
+    title = db.Column(db.String)
+    description = db.Column(db.String)
+    completed_at = db.Column(db.DateTime, nullable = True)
+    goal_id = db.Column(db.Integer, db.ForeignKey("goal.goal_id"), nullable=True)
+    goal = db.relationship("Goal", back_populates = "tasks")
+
+    def to_dict_task(self):
+        if self.completed_at is not None:
+            completed = True
+        elif self.completed_at is None:
+            completed = False
+        if self.goal_id is None:
+
+            return {"task": {
+            "id":self.task_id, 
+            "title":self.title, 
+            "description": self.description,
+            "is_complete": completed
+            }}
+        else: #self.goal_id is True ,aka this task has a goal
+            return {"task": {
+            "id":self.task_id, 
+            "goal_id": self.goal_id,
+            "title":self.title, 
+            "description": self.description,
+            "is_complete": completed
+            }}
+
+
+    def to_dict_all_tasks(self):
+        return {
+        "id":self.task_id,
+        "title":self.title, 
+        "description": self.description,
+        "is_complete": False}
```

Comment:
Small bug here, the "is_complete" param in this function and below on line 47 should follow the pattern from `to_dict_task` (ie, False if `completed_at` is None, otherwise True).